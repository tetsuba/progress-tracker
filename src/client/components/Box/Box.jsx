import React from 'react'

export default function Box({ children, max }) {
  const style = {
    maxWidth: max,
    width: '100%',
    padding: '20px 20px',
    backgroundColor: '#f8f9fa',
    margin: '0 auto',
    borderRadius: '8px',
    border: 'solid 1px #ccc',
  }

  return <div style={style}>{children}</div>
}
