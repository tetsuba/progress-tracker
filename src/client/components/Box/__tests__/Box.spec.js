import { testRendererFull } from '../../../../test/testHelper'
import Box from '../Box'

describe('<Box>', () => {
  it('should render with a max width of 500px', () => {
    const props = {
      max: 500,
    }
    const wrapper = testRendererFull(Box, props)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render with a class name', () => {
    const props = {
      max: 500,
      className: 'mt-5',
    }
    const wrapper = testRendererFull(Box, props)
    expect(wrapper).toMatchSnapshot()
  })
})
