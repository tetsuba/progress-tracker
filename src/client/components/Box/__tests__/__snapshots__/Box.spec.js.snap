// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box> should render with a class name 1`] = `
<Box
  className="mt-5"
  max={500}
>
  <div
    className="mt-5"
    style={
      Object {
        "backgroundColor": "#f8f9fa",
        "border": "solid 1px #ccc",
        "borderRadius": "8px",
        "margin": "0 auto",
        "maxWidth": 500,
        "padding": "20px 20px",
        "width": "100%",
      }
    }
  />
</Box>
`;

exports[`<Box> should render with a max width of 500px 1`] = `
<Box
  max={500}
>
  <div
    style={
      Object {
        "backgroundColor": "#f8f9fa",
        "border": "solid 1px #ccc",
        "borderRadius": "8px",
        "margin": "0 auto",
        "maxWidth": 500,
        "padding": "20px 20px",
        "width": "100%",
      }
    }
  />
</Box>
`;
